src/cli/controller.js

```diff
@@ -114,6 +114,10 @@ class Controller {
 
         return new Promise((resolve, reject) => {
           // Parse the user's code to find the names of the exported functions
+          if (opts.firebase) {
+            // If call is coming from Firebase, assume functions have already been parsed
+            return resolve(name.replace(/\-/g, '.'));
```

Comment:
I don't  think you have to do the replace, since the Firebase CLI would have already done the dash to dot substitution during the trigger parsing step. Can replace this line with:
return resolve(name)

But you should test to make sure.

Reply:
The CLI does not do the dot substitution. We could also go with opts.entryPoint here, instead of replacing the dashes with dots in the name.

Reply:
Actually, I got confused. What I meant is that it is fine to leave the name with dashes in them. You do not need to put the dots back. When the Firebase CLI deploys functions to the real GCF API, the name of each function stays the  "dashed" version, since the API doesn't actually allow dots in the name of functions. So you should still replace this line with: `return resolve(name)`